pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -2243,6 +2243,176 @@ def is_hermitian(self):
         return True
 
 
+class IntegerComparator(Operation):
+    r"""IntegerComparator(value, geq, control_wires, wires)
+    Apply a Pauli X gate controlled on the following condition:
+
+    Given a basis state :math:`\vert \sigma \rangle`, where :math:`\sigma \in \mathbb{N}`, and a fixed positive
+    integer :math:`L`, flip a target qubit if :math:`\sigma \geq L`. Alternatively, the flipping condition can
+    be :math:`\sigma < L`.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 0
```

Comment:
This is not exactly true. If set to `0` to not have issues with differentiable, then worth having a note box in this docstring about that to inform users.